src/equations/compressible_navier_stokes_2d.jl

```diff
@@ -0,0 +1,338 @@
+  varnames(variable_mapping, equations_parabolic.equations_hyperbolic)
+
+# we specialize this function to compute gradients of primitive variables instead of
+# conservative variables.
+gradient_variable_transformation(::CompressibleNavierStokesEquations2D, dg_parabolic) = cons2prim
+
+# no orientation specified since the flux is vector-valued
+# Explicit formulas for the diffussive Navier-Stokes fluxes are available, e.g. in Section 2
+# of the paper by Svärd, Carpenter and Nordström
+# "A stable high-order finite difference scheme for the compressible Navier–Stokes
+#  equations, far-field boundary conditions"
+# Although these authors use a different nondimensionalization so some constants are different
+# particularly for Fick's law.
+#
+# Note, could be generalized to use Sutherland's law to get the molecular and thermal
+# diffusivity
+function flux(u, grad_u, equations::CompressibleNavierStokesEquations2D)
+  # Here grad_u is assumed to contain the gradients of the primitive variables (v1, v2, T)
+  # either computed directly or reverse engineered from the gradient of the entropy vairables
+  # by way of the `convert_gradient_variables` function
+  rho, v1, v2, _ = u
+
+  # grad_u contains derivatives of each hyperbolic variable
+  _, dv1dx, dv2dx, dTdx = grad_u[1]
+  _, dv1dy, dv2dy, dTdy = grad_u[2]
+
+  # Components of viscous stress tensor
+
+  # (4/3 * (v1)_x - 2/3 * (v2)_y)
+  tau_11 = ( 4.0 / 3.0 * dv1dx - 2.0 / 3.0 * dv2dy )
+  # ((v1)_y + (v2)_x)
+  # stress tensor is symmetric
+  tau_12 = ( dv1dy + dv2dx ) # = tau_21
+  # (4/3 * (v2)_y - 2/3 * (v1)_x)
+  tau_22 = ( 4.0 / 3.0 * dv2dy - 2.0 / 3.0 * dv1dx )
+
+  # Fick's law q = -kappa * grad(T); constant is kappa = gamma μ R / ((gamma-1) Pr)
+  # Important note! Due to nondimensional scaling R = 1 / gamma, so the
+  # temperature T in the gradient computation already contains a factor of gamma
+  q1 = equations.kappa * dTdx
+  q2 = equations.kappa * dTdy
+
+  # kinematic viscosity is simply 1/Re for this nondimensionalization
+  mu = 1.0 / equations.Re
+
+  # viscous flux components in the x-direction
+  f1 = zero(rho)
+  f2 = tau_11 * mu
+  f3 = tau_12 * mu
+  f4 = ( v1 * tau_11 + v2 * tau_12 + q1 ) * mu
+
+  # viscous flux components in the y-direction
+  # Note, symmetry is exploited for tau_12 = tau_21
+  g1 = zero(rho)
+  g2 = f3 # tau_21 * mu
+  g3 = tau_22 * mu
+  g4 = ( v1 * tau_12 + v2 * tau_22 + q2 ) * mu
+
+  return (SVector(f1, f2, f3, f4) , SVector(g1, g2, g3, g4))
+end
+
+
+# Convert conservative variables to primitive
+@inline function cons2prim(u, equations::CompressibleNavierStokesEquations2D)
+  rho, rho_v1, rho_v2, _ = u
+
+  v1 = rho_v1 / rho
+  v2 = rho_v2 / rho
+  T  = temperature(u, equations)
+
+  return SVector(rho, v1, v2, T)
+end
+
+# TODO: make this consistent with cons2prim above and cons2prim for Euler!
+@inline prim2cons(u, equations::CompressibleNavierStokesEquations2D) =
+    prim2cons(u, equations.equations_hyperbolic)
```

Comment:
Is this used anywhere? If not, I might suggest that we remove it.

Reply:
I believe this is a vestige of my misinterpretation that `CompressibleNavierStokesEquations2D` would carry its own `vars` and `nvars`. As far as I can tell this is unused and can be removed.